diff --git a/src/lib/arch/unix/ArchPluginUnix.cpp b/src/lib/arch/unix/ArchPluginUnix.cpp
index 997c274..3e390f0 100644
--- a/src/lib/arch/unix/ArchPluginUnix.cpp
+++ b/src/lib/arch/unix/ArchPluginUnix.cpp
@@ -76,8 +76,8 @@ ArchPluginUnix::load()
 		void* library = dlopen(path.c_str(), RTLD_LAZY);
 
 		if (library == NULL) {
-			LOG((CLOG_ERR "failed to load plugin: %s", (*it).c_str()));
-			throw XArch(dlerror());
+			LOG((CLOG_ERR "failed to load plugin '%s', error: %s", (*it).c_str(), dlerror()));
+			continue;
 		}
 
 		String filename = synergy::string::removeFileExt(*it);
